compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -450,13 +451,31 @@ class ExtractSemanticDB extends Phase:
         props |= SymbolInformation.Property.ENUM.value
       props
 
+    private def symbolAccess(sym: Symbol, kind: SymbolInformation.Kind)(using Context): Access =
+      kind match
+        case k.LOCAL | k.PARAMETER | k.SELF_PARAMETER | k.TYPE_PARAMETER | k.PACKAGE | k.PACKAGE_OBJECT =>
+          Access.Empty
+        case _ =>
+          if (sym.privateWithin == NoSymbol)
+            if (sym.isAllOf(PrivateLocal)) PrivateThisAccess()
+            else if (sym.is(Private)) PrivateAccess()
+            else if (sym.isAllOf(ProtectedLocal)) ProtectedThisAccess()
+            else if (sym.is(Protected)) ProtectedAccess()
+            else PublicAccess()
+          else
+            val ssym = symbolName(sym.privateWithin)
+            if (sym.isPrivate) PrivateWithinAccess(ssym)
```

Comment:
I think here that a private qualified symbol never gets the private flag, you should instead check if it is protected. See the other comment about the `metac.expect` output for `tests/semanticdb/expect/Access.scala`

Reply:
You're right, fixed in https://github.com/lampepfl/dotty/pull/12964/commits/fb6b679db92e46e8517f6590be4771d6f331ec34 👍